lib/utils/bucket-path-prefix-helper.js

```diff
@@ -0,0 +1,39 @@
+/* @flow */
+'use strict';
+
+function getFileFromKey (
+  bucketPathPrefix /* : string */,
+  key /* : string */
+) {
+  let file = key;
+  if (bucketPathPrefix) {
+    // bucketPathPrefix.length - 1 as it will end '/'
+    file = file.substr(bucketPathPrefix.length);
+  }
+  return file;
+}
+
+function validate (
+  bucketPathPrefix /* : string | void */
+) /* : string */ {
+  if (!bucketPathPrefix) {
+    return '';
+  }
+
+  // prefix should have a slack at the end or be an empty string.
+  if (!bucketPathPrefix.endsWith('/')) {
```

Comment:
this requires node 6.10.3 (which is the current LTS, about 2 weeks old), but `package.json` says it needs node 4. Im happy using the function if we bump the `package.json` version

also its called a "slash" not a "slack" :P